modules/currency-l0/src/main/scala/org/tessellation/currency/http/routes/CurrencyRoutes.scala

```diff
@@ -0,0 +1,84 @@
+package org.tessellation.currency.http.routes
+
+import cats.effect.Async
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import org.tessellation.currency.domain.cell.{L0Cell, L0CellInput}
+import org.tessellation.currency.schema.currency.{CurrencyBlock, CurrencySnapshot}
+import org.tessellation.ext.http4s.AddressVar
+import org.tessellation.sdk.ext.http4s.SnapshotOrdinalVar
+import org.tessellation.security.signature.Signed
+import io.circe.shapes._
+import org.http4s.HttpRoutes
+import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
+import org.http4s.dsl.Http4sDsl
+import org.http4s.server.Router
+import org.tessellation.sdk.domain.snapshot.services.AddressService
+import shapeless._
+import shapeless.syntax.singleton._
+
+final case class CurrencyRoutes[F[_]: Async](addressService: AddressService[F, CurrencySnapshot], mkCell: L0Cell.Mk[F])
+    extends Http4sDsl[F] {
+  private[routes] val prefixPath = "/dag"
```

Comment:
probably not `/dag`. by the way. I think Routes can be abstracted (possibly as another PR), as it depends on an address service.
Also, I'd change `/dag` to `/currency` and make an alias `/dag` -> `/currency` (or redirection) in our core module, to make it an universal endpoint.

Reply:
Alias redirection is safer because we won't break existing integration 👍🏻

Reply:
Also I will extract `CurrencyRoutes` because it is 1:1 with DAG routes. Let's make it in next PR.